### PR TITLE
Feat: add blobs to eth history

### PR DESCRIPTION
**Pull request by allnil**
draft for: #6330

### CODE REVIEW

File: crates/rpc/rpc/src/eth/api/fee_history.rs [hunk 1]
```diff
@@ -342,6 +348,9 @@ impl FeeHistoryEntry {
         FeeHistoryEntry {
             base_fee_per_gas: block.base_fee_per_gas.unwrap_or_default(),
             gas_used_ratio: block.gas_used as f64 / block.gas_limit as f64,
+            base_fee_per_blob_gas: block.blob_fee(),
+            blob_gas_used_ratio: block.blob_gas_used() as f64 /
+                reth_primitives::constants::eip4844::MAX_DATA_GAS_PER_BLOCK as f64,
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/api/fee_history.rs:353]: this looks correct to me

**Review by mattsse — CHANGES_REQUESTED** [review]
nice,

we're missing one thing:

>  title: baseFeePerBlobGasArray
          description: An array of block base fees per blob gas. This includes the next block after the newest of the returned range, because this value can be derived from the newest block. Zeroes are returned for pre-EIP-4844 blocks.

Same goes for `base_fee_per_gas` which we include for the next block:

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/api/fees.rs#L149-L154

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/api/fees.rs#L203-L208

File: crates/rpc/rpc/src/eth/api/fee_history.rs (hunk 1 above)
Comment (allnil) [reply crates/rpc/rpc/src/eth/api/fee_history.rs:353]: used as a reference:
https://github.com/NethermindEth/nethermind/blob/d6bf96493c11a487e0af69b2c6fba323d1216b05/src/Nethermind/Nethermind.JsonRpc/Modules/Eth/FeeHistory/FeeHistoryOracle.cs#L64

**Comment by mattsse** [comment]
fyi https://github.com/paradigmxyz/reth/blob/e085f4524f75ca3a52e86b1a46f92cdcf074f652/crates/primitives/src/header.rs#L242-L242

**Review by mattsse — CHANGES_REQUESTED** [review]
awesome, 
I realize that there's one thing missing here, the base_fee_per_blob_gas for the next block if we're having cached entries.

The FeeEntry currently doesn't have the required data to derive that for the next block

what we can do instead is storing the excess blob gas and the blob gas used in the cache entry

https://github.com/paradigmxyz/reth/blob/main/crates/primitives/src/header.rs#L258-L263

and compute the relevant fields via functions on the cache entry
https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/api/fee_history.rs#L351-L351

this way we can also derive the values for the next block
https://github.com/paradigmxyz/reth/blob/main/crates/primitives/src/header.rs#L261-L262

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**